rusoto/credential/src/instance_metadata.rs

```diff
@@ -36,10 +36,29 @@ const AWS_CREDENTIALS_PROVIDER_PATH: &str = "latest/meta-data/iam/security-crede
 ///   // ...
 /// }
 /// ```
+///
+/// The source location can be changed from the default of 169.254.169.254:
+///
+/// ```rust
+/// extern crate rusoto_credential;
+///
+/// use std::time::Duration;
+///
+/// use rusoto_credential::InstanceMetadataProvider;
+///
+/// fn main() {
+///   let mut provider = InstanceMetadataProvider::new();
+///   // you can overwrite the default endpoint like this:
+///   provider.set_ip_addr_with_port("127.0.0.1", "8080");
+///
+///   // ...
+/// }
+/// ```
 #[derive(Clone, Debug)]
 pub struct InstanceMetadataProvider {
     client: HttpClient,
     timeout: Duration,
+    metadata_ip_addr: String,
```

Comment:
The most recent change moved this field to a `String` type. This simplifies the new behavior at the cost of some `clone`s. Attempts to convert the type to `&str` led to a lot of back and forth with the borrow checker and I feel it's an improvement we can address later: the win of having an integration test of the metadata provider is bigger than the potential runtime performance loss.

Instance metadata shouldn't be called super often anyways. 😄